python/sbp/client/util/settingmonitor.py

```diff
@@ -12,21 +12,41 @@
 monitor SBP_MSG_SETTINGS_READ_RESP messages for desired values
 """
 
+from sbp.settings import SBP_MSG_SETTINGS_READ_RESP
+
 import time
 
+
 class SettingMonitor(object):
+    """Class to monitor Settings via SBP messages
+
+    Parameters
+    ----------
+    None
+
+    """
+
     def __init__(self):
         self.settings = []
 
     def capture_setting(self, sbp_msg, **metadata):
-        section, setting, value = sbp_msg.payload.split('\0')[:3]
-        self.settings.append((section,setting,value))
+        """Callback to extract and store setting values from
+        SBP_MSG_SETTINGS_READ_RESP
+
+        Messages of any type other than SBP_MSG_SETTINGS_READ_RESP are ignored
+        """
+        if sbp_msg.msg_type == SBP_MSG_SETTINGS_READ_RESP:
+            section, setting, value = sbp_msg.payload.split('\0')[:3]
+            self.settings.append((section, setting, value))
 
     def wait_for_setting_value(self, section, setting, value, wait_time=5.0):
+        """Function to wait wait_time seconds to see a
+        SBP_MSG_SETTINGS_READ_RESP message with a user-specified value
+        """
         expire = time.time() + wait_time
         ok = False
         while not ok and time.time() < expire:
-            settings = filter(lambda x: (x[0],x[1]) == (section, setting),
+            settings = filter(lambda x: (x[0], x[1]) == (section, setting),
```

Comment:
This sort of depends on the intended usage, but it seems like this would make it so that you couldn't do something like:
MsgSettingsReadReq msg1
MsgSettingsReadReq msg2
...
wait_for_setting_value msg1
wait_for_setting_value msg2

since it's possible that you'd end up clearing the second message when you call the first wait. It seems like coupling the request and the wait together makes sense. Alternatively, you could just clear the messages that match the requested type and use different logic for checking if they're in the buffer.

Reply:
The settings/messages aren't cleared in the wait (`settings` -vs- `self.settings`). That needs to happen explicitly.

However, after looking more at this logic, I'm liking the part about looking at the last message less and less. If the user wants to see the "last" message, then they can explicitly clear the messages. I like your idea of adding the ability to clear only certain messages.

Reply:
Hmm, I still think we need that "last" message logic in there at some point though, since there may well be multiple messages of the same type in there.

Reply:
@axlan I think the last message logic should stay in there, but I'm open to suggestions. I've added the ability to selectively clear the settings based on a combination of section/setting/value.